config/initializers/devise.rb

```diff
@@ -286,30 +286,42 @@
         "#{host}/users/auth/cis2/callback"
       end
 
-    config.omniauth :openid_connect,
-                    {
-                      name: :cis2,
-                      scope: %i[
-                        openid
-                        profile
-                        email
-                        nationalrbacaccess
-                        professionalmemberships
-                      ],
-                      response_type: :code,
-                      # uid_field: "preferred_username",
-                      issuer: Settings.cis2.issuer,
-                      discovery: true,
-                      client_auth_method: :jwks,
-                      client_options: {
-                        port: 443,
-                        scheme: "https",
-                        host: Settings.cis2.host,
-                        identifier: Settings.cis2.client_id,
-                        secret: Settings.cis2.secret,
-                        redirect_uri:
-                      }
-                    }
+    config.omniauth(
+      :openid_connect,
+      {
+        setup:
+          lambda do |env|
+            if env.dig("rack.request.form_hash", "change_role")
+              env["omniauth.strategy"].options["scope"] << :changedrole
+              env["omniauth.strategy"].options["prompt"] = :login
+            else
+              env["omniauth.strategy"].options["scope"] << :selectedrole
+            end
+          end,
```

Comment:
I'd consider moving this to a variable above like `redirect_uri`, I think it'd be slightly cleaner